craft_parts/plugins/cargo_package_plugin.py

```diff
@@ -0,0 +1,162 @@
+
+    properties_class = CargoPackagePluginProperties
+    validator_class = CargoPackagePluginEnvironmentValidator
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build environment."""
+        return set()
+
+    @property
+    def _registry_dir(self) -> pathlib.Path:
+        return self._part_info.project_info.dirs.stage_dir / "cargo_registry"
+
+    @property
+    def _registry_output_dir(self) -> pathlib.Path:
+        return self._part_info.part_install_dir / "cargo_registry"
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        return {}
+
+    def _get_package_command(self) -> str:
+        options = cast(CargoPackagePluginProperties, self._options)
+        cargo = options.cargo_package_cargo_command
+        features = ",".join(options.cargo_package_features)
+        if features:
+            features = f"--features {features}"
+        return f"{cargo} package --no-verify {features}"
+
+    @override
+    def get_build_commands(self) -> list[str]:
+        """Return a list of commands to run during the build step."""
+        # We do not want this implementation detail exposed in the run script
+        # This sets the default registry for all parts to use.
+        cargo_config = pathlib.Path("~/.cargo/config.toml").expanduser().resolve()
+        if not cargo_config.exists():
+            cargo_config.parent.mkdir(exist_ok=True, parents=True)
+            cargo_config.write_text(
+                CARGO_TOML_TEMPLATE.format(registry_dir=self._registry_dir)
+            )
+
+        # This is a bit hard to read as it's a combination of pythonisms and bashisms.
+        # What we're doing here is creating a bash clause that looks up the value
+        # of the CARGO_REGISTRY_DIRECTORY environment variable and, if it's not
+        # assigned, uses the plugin's default registry_output_dir. This way,
+        # the user can replace the registry directory if they want. For example,
+        # to use the Debian cargo registry at /usr/share/cargo/registry, they
+        # can set CARGO_REGISTRY_DIRECTORY=/usr/share/cargo/registry and then in
+        # all other parts set the cargo config environment variable
+        # CARGO_REGISTRY_DEFAULT to the value "apt"
+        write_registry_bash = (
+            f"${{CARGO_REGISTRY_DIRECTORY:-{self._registry_output_dir}}}"
+        )
+
+        return [
+            self._get_package_command(),
+            textwrap.dedent(
+                f"""\
+                    pushd target/package
+                    for package in *.crate; do
+                        tar xf $package
+                        rm $package
```

Comment:
I don't expect spaces in package names but may as well:

```suggestion
                        tar xf "$package"
                        rm "$package"
```